crates/matrix-sdk/src/room/common.rs

```diff
@@ -236,6 +239,22 @@ impl Common {
             response.chunk.extend(http_response.chunk.into_iter().map(TimelineEvent::new));
         }
 
+        let Some(push_context) = self.push_context().await? else {
+            return Ok(response);
+        };
+
+        let push_rules = self
+            .client()
+            .account()
+            .account_data::<PushRulesEventContent>()
+            .await?
+            .and_then(|r| r.deserialize().ok().map(|r| r.global))
+            .unwrap_or_else(|| Ruleset::server_default(self.own_user_id()));
+
+        for event in &mut response.chunk {
+            event.push_actions = push_rules.get_actions(&event.event, &push_context).to_owned();
+        }
```

Comment:
I think the little bit of extra indentation you get when making this an `if let` isn't too bad, I'd prefer that over having two return places that return the same expression.